### DISCUSSION TITLE
How to scale up fastapi application?

### DISCUSSION THREAD

**Discussion by sivanthisuresh** [fastapi/fastapi#3011] [Questions]
Is fastapi suitable for application like online polling application where Millions of user perform post request to single endpoint running on single server?

**Comment by krishnardt** [comment]
Yes.. it can handle.
And handling the network is totally dependent upon memory of the server.

If you are using only one server, vertical scaling of memory and no of cores can help you to increase the request handling.

By reducing the data transfer between the api and client calls(json not preferable in few cases), and  Managing the load and query structure on database with optimised logic will ensure lesser load helps you to reach your goal.
